src/modules/common/helpers/deep.clone.helper.ts

```diff
@@ -16,7 +16,7 @@ const deepClone = (data) => {
         return data;
     }
 
-    if (Object.prototype.toString.apply(data) === '[object Array]') {
+    if (isArray(data)) {
```

Comment:
Here we needed to add custom type guard (https://www.typescriptlang.org/docs/handbook/advanced-types.html#type-guards-and-differentiating-types)
so that we have type checking for array inside `if` statement.
We do not need custom guards in case of simpler conditions such as `data instanceof Array`.